app/src/main/java/org/simple/clinic/registration/phone/PhoneNumberValidator.kt

```diff
@@ -13,11 +14,11 @@ interface PhoneNumberValidator {
     MOBILE
   }
 
-  enum class Result {
-    VALID,
-    LENGTH_TOO_SHORT,
-    LENGTH_TOO_LONG,
-    BLANK
+  sealed class Result {
+    object ValidNumber : Result()
+    data class LengthTooShort(val limit: Int) : Result()
+    data class LengthTooLong(val limit: Int) : Result()
```

Comment:
Instead of `limit`, can we use more appropriate names which indicate what they are?

Ex: `mininumRequiredNumberLength` and `maximumAllowedNumberLength`.